tsl/src/remote/connection.c

```diff
@@ -289,7 +289,15 @@ fill_result_error(TSConnectionError *err, int errcode, const char *errmsg, const
 	const char *sqlstate;
 
 	if (NULL == err || NULL == res || NULL == entry)
+	{
+		if (NULL != err)
```

Comment:
I always have a mind break when read this type of comparison between value and variable, why not use just:

```C
if (!err || !res || !entry)
{
    if (err)
...
```

I think we don't have a clear definition for this but to me is weird and @svenklemm also complained about it in other PR that I worked.

Reply:
I guess this is a preference of style. 

While I agree that the suggested change is somewhat easier on the eye, it is also somewhat ambiguous. 

By explicitly comparing with `NULL` one knows that the variables are pointers without having to look at the variable definition (which might be outside this context, especially true in code-review snippets).

Let's take the variable `err` for example. From just reading the code, it could easily be an `int` , a `boolean`, a pointer, or even an `enum` type. So, by writing:

```
if (NULL == err)
```
I immediately know that `err` is a pointer and this is a `NULL` check. Similarly, if I see this:

```
if (0 == err)
```
I know err is an integer and the non-error condition is 0. Or I see this to identify enum types"

```
if (NO_ERROR == err)
```

However, if I see this:

```
if (!err)
```

then I don't know the type of `err` (is it int, pointer, bool, or even an enum type?) and I don't know if, e.g., 0 is the non-error condition. 

Just explaining some reasoning here. Don't feel super-strongly about it though, and I am sure we have a mix of styles in the code.

Reply:
Personally, I'm fine with any code style. In this case, I just followed the style that was already in use.

Reply:
Understood your point. Can we at least invert the comparison??? 

```C
if (err == NULL)
```

IMHO is more clear compare a variable with a value.

Reply:
There's actually a point to this too :)

writing 

```
if (NULL == err)
```

protects against accidental assignment, e.g.:

```
if (err = NULL)
```

Most compilers today detect that case, however.

Reply:
@fabriziomello this is very a common approach in C world 🤷‍♂️ Again, I simply followed the code style that was chosen in the project (or this particular part of it) a long time before I joined the project. If you believe the code style can be improved, I'm sure we all are open to suggestions. However, this should be discussed not in a separate pull request.

Reply:
Aaaand it's gone http://www.failure.at/wp-content/uploads/2013/07/4387467_460s.jpg

Reply:
@fabriziomello source: https://en.wikipedia.org/wiki/Yoda_conditions